### PR TITLE
Fixes #24852 - Creating Host with empty allocation fails

### PR DESCRIPTION
**Pull request by apuntamb**
In case of clicking on Size, the Allocation text field becomes editable and user can give any custom size as per their requirement. But, I think keeping it empty is as good as "0G" for the user. Therefore, I have set the allocation to 0G in case when user selects Size option. 
Any suggestions ?
Also, creating a PR in the fog/fog-libvirt repo to handle when user sends empty allocation.

### CODE REVIEW

**Comment by apuntamb** [comment]
The fog/fog-libvirt PR is https://github.com/fog/fog-libvirt/pull/62

File: app/models/compute_resources/foreman/model/libvirt.rb
Comment (xprazak2) [inline app/models/compute_resources/foreman/model/libvirt.rb:282]: You will append 'G' if the allocation already ends with it, resulting in:

**Comment by xprazak2** [comment]
It also fails with `undefined method '[]' for nil:NilClass` if I submit with a random string as allocation. Any chance that could be fixed as well?

**Comment by ohadlevy** [comment]
whats the status of this pr? thanks

**Comment by apuntamb** [comment]
@ohadlevy Hello. Thanks for pinging on this. Will inspect this and update the PR soon.

**Comment by xprazak2** [comment]
@apuntamb any progress on this?

**Comment by apuntamb** [comment]
@xprazak2 Sorry for the delay. Let me get to this today.

File: app/models/compute_resources/foreman/model/libvirt.rb
Comment (apuntamb) [reply app/models/compute_resources/foreman/model/libvirt.rb]: @xprazak2 This should now solve the problem of `GG`.
For `undefined method [] for nil:NilClass` error, I've tried to make the change in [#fog-libvirt/62](https://github.com/fog/fog-libvirt/pull/62). Can we once again review the PRs together to see if this still leads to any issue?

Comment (xprazak2) [reply app/models/compute_resources/foreman/model/libvirt.rb]: Works well with the fog patch.

Comment (apuntamb) [reply app/models/compute_resources/foreman/model/libvirt.rb]: [#fog-libvirt/62](https://github.com/fog/fog-libvirt/pull/62) has now been merged.

Comment (xprazak2) [reply app/models/compute_resources/foreman/model/libvirt.rb]: Nice work! I asked maintainer for a new release. I would wait with merge until we have the new gem version so that we do not forget to get it into packaging.

Comment (apuntamb) [reply app/models/compute_resources/foreman/model/libvirt.rb]: Sure, makes sense. We'll wait for their new release.

**Comment by xprazak2** [comment]
We now have a new version of fog-libvirt with the fix, I think we can merge now, let's see if tests still pass.

**Comment by xprazak2** [comment]
[test foreman]

**Comment by xprazak2** [comment]
[test foreman]

**Comment by xprazak2** [comment]
@apuntamb, could you please rebase to see if tests still pass now that we have newer forg-libvirt?

**Comment by mmoll** [comment]
probably only needs a rebase and a Rubocop fix

**Comment by apuntamb** [comment]
@xprazak2 @mmoll Rebased. I'm not sure what Rubocop fix would be required here.

File: app/models/compute_resources/foreman/model/libvirt.rb
Comment (mmoll) [reply app/models/compute_resources/foreman/model/libvirt.rb]: @apuntamb Rubocop error :point_up:

Comment (apuntamb) [reply app/models/compute_resources/foreman/model/libvirt.rb]: thanks. done

**Comment by apuntamb** [comment]
@xprazak2 @mmoll Fixed test failures too.

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks everyone!

**Pull request merged**
